### PR TITLE
Add publish time and storage stats benchmarks for multiple batch sizes

### PR DESCRIPTION
**Pull request by eozturk1**
Sample run for publishing with different batch sizes:

```
Starting micro-benchmarks:
  2812.80 +/- 54.75 ms ...............................create notification
   623.20 +/- 24.46 ms ...............................verify notification
     0.00 +/- 0.00  ms .......................................create vote
     0.00 +/- 0.00  ms .......................................verify vote
     0.00 +/- 0.00  ms .............................aggregate certificate
     1.00 +/- 0.00  ms ................................verify certificate
     2.20 +/- 0.42  ms ....................publish_batch_size_1_in_memory
    23.30 +/- 1.95  ms ...................publish_batch_size_10_in_memory
   223.00 +/- 3.30  ms ..................publish_batch_size_100_in_memory
  2398.10 +/- 43.99 ms .................publish_batch_size_1000_in_memory
    32.50 +/- 10.26 ms ...................publish_batch_size_1_persistent
    51.20 +/- 6.30  ms ..................publish_batch_size_10_persistent
   264.90 +/- 11.73 ms .................publish_batch_size_100_persistent
  2625.70 +/- 57.50 ms ................publish_batch_size_1000_persistent
```

RocksDB storage stats sample:

```
** DB Stats **
Uptime(secs): 0.0 total, 0.0 interval
Cumulative writes: 6 writes, 6 keys, 6 commit groups, 1.0 writes per commit group, ingest: 0.00 GB, 0.07 MB/s
Cumulative WAL: 6 writes, 0 syncs, 6.00 writes per sync, written: 0.00 GB, 0.07 MB/s
Cumulative stall: 00:00:0.000 H:M:S, 0.0 percent
Interval writes: 6 writes, 6 keys, 6 commit groups, 1.0 writes per commit group, ingest: 0.00 MB, 0.07 MB/s
Interval WAL: 6 writes, 0 syncs, 6.00 writes per sync, written: 0.00 GB, 0.07 MB/s
Interval stall: 00:00:0.000 H:M:S, 0.0 percent

Memory usage stats: Mem table total: 2048, Mem table unflushed: 2048, Mem table readers total: 150, Cache total: 0
```

Questions:

- What is a good `DEFAULT_PRECISION` value to set? For 100, it takes a long time to run the benchmarks. Is only average of 10 runs acceptable?
- What batch sizes should we use for time + storage stats measurements? 10^x  where x is in [2, 5]?
- RocksDB's storage stats are limited. Is there anything else we are interested in from the [list](https://docs.rs/rocksdb/latest/rocksdb/properties/index.html) of stats? (The ones I printed were the ones I thought were most relevant.

To fix:
- [ ] RocksDB `flush()` is ineffective. The memory stats still shows unflushed data.

### CODE REVIEW

**Review by asonnino — APPROVED** [review]

**Comment by eozturk1** [comment]
It seems I don't have write access so I can't merge this change. How should we proceed? Should we merge this to `main` or to a different branch?

**Pull request merged**
